sse2neon.h

```diff
@@ -1182,7 +1223,22 @@ FORCE_INLINE __m128i _mm_shuffle_epi32_default(__m128i a,
 
 // FORCE_INLINE __m128i _mm_shufflehi_epi16(__m128i a, __constrange(0,255) int
 // imm)
+#if defined(__clang__)
+#define _mm_shufflehi_epi16(a, imm)                      \
+    ({                                                   \
+         int16x8_t _input = vreinterpretq_s16_m128i(a);  \
+         int16x8_t _shuf =                               \
+              __builtin_shufflevector(_input, _input,    \
+                0, 1, 2, 3,                              \
+                ((imm) & 0x3) + 4,                       \
+                (((imm) >> 2) & 0x3) + 4,                \
+                (((imm) >> 4) & 0x3) + 4,                \
+                (((imm) >> 6) & 0x3) + 4);               \
```

Comment:
The addition of 4 is not really necessary.

Reply:
Yes it is, it is an 8 element vector and it is selecting the high lanes.

Reply:
Oh my bad again.